### PR TITLE
Move integration test base code to subproject

### PR DESCRIPTION
**Pull request by brfrn169**
After #661, we are now able to dynamically load storage adapters and add a new storage adapter in another project. And, the next task is to make the integration test be able to run in another project.

To do that, we need to move the integration test base code to a subproject and publish the integration test jar to the Maven repository.

What I did in this change is as follows:

- Created `integration-test` subproject
- Moved the integration test base code (from `core` and `schema-loader`) to the `integration-test` subproject
- Moved the integration test code in `schema-loader` to `core`
- Did necessary refactoring for the integration test code to move it to the `integration-test` subproject

I will also add inline comments for the details. Please take a look!

### CODE REVIEW

File: server/src/integration-test/java/com/scalar/db/server/ServerAdminTestUtils.java [hunk 1]
```diff
@@ -9,21 +9,24 @@
...
+import com.scalar.db.util.AdminTestUtils;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import java.sql.Connection;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.Properties;
 import org.apache.commons.dbcp2.BasicDataSource;
 
-public class JdbcAdminTestUtils extends AdminTestUtils {
+public class ServerAdminTestUtils extends AdminTestUtils {
```
Comment (brfrn169) [inline server/src/integration-test/java/com/scalar/db/server/ServerAdminTestUtils.java:20]: A `XXXAdminTestUtils` class for server.

File: server/src/integration-test/java/com/scalar/db/server/ServerEnv.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package com.scalar.db.server;
```
Comment (brfrn169) [inline server/src/integration-test/java/com/scalar/db/server/ServerEnv.java:1]: Changed to use JDBC in the server integration tests.

**Review by Torch3333 — DISMISSED** [review]
LGTM, thank you!

File: core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+public class CassandraAdminTestUtils extends AdminTestUtils {
+
+  public CassandraAdminTestUtils(Properties properties) {
+    super(properties);
+  }
+
+  @Override
+  public void dropMetadataTable() {
+    // Do nothing
```
Comment (feeblefakie) [inline core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java:14]: Just a question.
I thought the metadata table is now managed for Cassandra as well for consistency, is it OK with empty?

**Review by feeblefakie — DISMISSED** [review]
LGTM! Thank you!
Left one question but doesn't block it from being merged.

File: core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java (hunk 3 above)
Comment (brfrn169) [reply core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java:14]: Thank you for taking a look! 

The change is not yet merged to the `master` branch. In the `feat/namespaces_management` branch, the metadata table is managed for Cassandra.

Comment (Torch3333) [reply core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java:14]: Maybe there is some confusion but the `feat/namespaces_management` only implements the namespace management for Cassandra. 
Managing the table metadata using a metadata table for Cassandra is not yet implemented.

Comment (feeblefakie) [reply core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java:14]: Thank you for clarification!
So, I understand `Do nothing` parts will eventually be implemented after the table metadata for Cassandra is implemented.

Comment (Torch3333) [reply core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminTestUtils.java:14]: Yes, exactly.

**Pull request merged**